packages/toolkit/src/listenerMiddleware/tests/effectScenarios.test.ts

```diff
@@ -57,10 +54,7 @@ describe('Saga-style Effects Scenarios', () => {
     return new Promise((resolve) => setTimeout(resolve, ms))
   }
 
-  beforeAll(() => {
-    const noop = () => {}
-    vi.spyOn(console, 'error').mockImplementation(noop)
-  })
+  const consoleErrorSpy = vi.spyOn(console, 'error').mockImplementation(noop)
```

Comment:
I just wanted to suggest that we do 
```suggestion
  using consoleErrorSpy = vi.spyOn(console, 'error').mockImplementation(noop)
```
inside the tests that actually need this, and skip the `mockClear` and `mockRestore` calls, but I'm not sure if `vitest` actually returns a `Disposable` here. I know `jest` does (because I added the functionality there) - could you check if `vitest` has that now, too?

Reply:
Sure I can try it. The type tests will probably fail, so I'll probably exclude the runtime tests from the type tests.

Reply:
Looks like `vitest` doesn't return  a `Disposable`, though I might try to submit a PR to add that functionality for `vitest`.

Reply:
Will definitely be a win in the long run, even if it's not applicable to this PR in the end :)